untitled1/src/main/com/aca/homework/week9/expressioncalculator/SimpleCalculator.java

```diff
@@ -0,0 +1,48 @@
+package com.aca.homework.week9.expressioncalculator;
+
+import java.util.Scanner;
+
+public class SimpleCalculator {
+    private StringSupplier stringSupplier;
+
+    public SimpleCalculator(StringSupplier stringSupplier) {
+        this.stringSupplier = stringSupplier;
+    }
+
+    public String askAndCalculate() {
+        String expression = stringSupplier.get();
+
+        int lengthOfExpression = expression.length();
+        if (lengthOfExpression != 3) {
+            System.out.println("The user input expression having other format then \' digit operation digit \' ");
+            return "error";
+        }
+
+        char operation = expression.charAt(1);
+        try {
+            OperationType operationType = OperationType.charConvertToOperationType(operation);
+            int firstOperand = convertCharToDigit(expression.charAt(0));
+            int secondOperand = convertCharToDigit(expression.charAt(2));
+
+            if (firstOperand != -1 && secondOperand != -1) {
+                String result = Double.toString(operationType.calculate(firstOperand, secondOperand));
+                System.out.println("result = " + result);
+                return result;
+            }
+        } catch (IllegalArgumentException e) {
```

Comment:
you can write the code without try-catch block too. just check if the exception is going to be thrown, return an error